### PR TITLE
Reduce logging noise when stepping down as master before state recovery

### PR DESCRIPTION
**Pull request by ywelsch**
Reduces the logging noise from the state recovery component when there are duelling elections.

Relates to #32006

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/gateway/GatewayService.java [hunk 1]
```diff
@@ -254,10 +250,15 @@ public void clusterStateProcessed(final String source, final ClusterState oldSta
             logger.info("recovered [{}] indices into cluster_state", newState.metaData().indices().size());
         }
 
+        @Override
+        public void onNoLongerMaster(String source) {
+            logger.debug("stepped down as master before recovering state [{}]", source);
```
Comment (andrershov) [inline server/src/main/java/org/elasticsearch/gateway/GatewayService.java]: Why do not we call `resetRecoveredFlags` here?

Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/gateway/GatewayService.java]: I think this needs a call to `resetRecoveredFlags()` too.

File: server/src/main/java/org/elasticsearch/gateway/GatewayService.java [hunk 2]
```diff
@@ -215,10 +215,8 @@ private void performStateRecovery(final boolean enforceRecoverAfterTime, final S
                 threadPool.generic().execute(new AbstractRunnable() {
                     @Override
                     public void onFailure(final Exception e) {
-                        logger.warn("Recovery failed", e);
-                        // we reset `recovered` in the listener don't reset it here otherwise there might be a race
-                        // that resets it to false while a new recover is already running?
-                        GatewayService.this.onFailure("state recovery failed: " + e.getMessage());
+                        logger.warn("state recovery failed", e);
+                        resetRecoveredFlags();
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/gateway/GatewayService.java]: Should we `resetRecoveredFlags()` on failure of the scheduled task a few lines above this too?

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
I requested one change and asked a question too.

File: server/src/main/java/org/elasticsearch/gateway/GatewayService.java (hunk 1 above)
Comment (ywelsch) [reply server/src/main/java/org/elasticsearch/gateway/GatewayService.java]: good catch!

Comment (ywelsch) [reply server/src/main/java/org/elasticsearch/gateway/GatewayService.java]: correct

File: server/src/main/java/org/elasticsearch/gateway/GatewayService.java (hunk 2 above)
Comment (ywelsch) [reply server/src/main/java/org/elasticsearch/gateway/GatewayService.java]: sure, added in 9de5c0e

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**